services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/storage.py

```diff
@@ -15,30 +16,58 @@
 _LIVENESS_TIMEOUT: Final[timedelta] = timedelta(seconds=5)
 
 
-async def _is_storage_responsive(url: str) -> bool:
+class _AuthTuple(NamedTuple):
+    username: str
+    password: str
+
+
+def _get_auth(storage_auth_settings: StorageAuthSettings) -> _AuthTuple | None:
+    if storage_auth_settings.auth_required:
+        assert storage_auth_settings.STORAGE_USERNAME  # nosec
+        assert storage_auth_settings.STORAGE_PASSWORD  # nosec
+        return _AuthTuple(
+            storage_auth_settings.STORAGE_USERNAME,
+            storage_auth_settings.STORAGE_PASSWORD.get_secret_value(),
+        )
+    return None
+
+
+def _get_url(storage_auth_settings: StorageAuthSettings) -> str:
+    url = URL(storage_auth_settings.api_base_url).with_path("/")
+    return f"{url}"
+
+
+async def _is_storage_responsive(storage_auth_settings: StorageAuthSettings) -> bool:
+    url = _get_url(storage_auth_settings)
```

Comment:
in't is more clear just inlining `storage_auth_settings.api_base_url` instead of creating yet another indirection for something so simple? Isnt' api_base_url not ending in `/`??

Reply:
The url for the healthcheck must end with a "/" and api_base_url is ending in `.../v0`.

I would tend it's more clear like this, since the name is pretty much straight forward and it avoids you having to repeat the same line of code twice.

Reply:
where are you using the same line of code? So far I see it only there :-)